packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart

```diff
@@ -3,12 +3,280 @@
+  final SKPaymentTransactionStateWrapper transactionState;
+
+  /// The payment that is created and added to the payment queue which generated this transaction.
+  final SKPaymentWrapper payment;
+
+  /// The original Transaction, only available if the [transactionState] is [SKPaymentTransactionStateWrapper.restored].
+  ///
+  /// When the [transactionState] is [SKPaymentTransactionStateWrapper.restored], the current transaction object holds a new
+  /// [transactionIdentifier].
+  final SKPaymentTransactionWrapper originalTransaction;
+
+  /// The timestamp of the transaction.
+  ///
+  /// Milliseconds since epoch.
+  /// It is only defined when the [transactionState] is [SKPaymentTransactionStateWrapper.purchased] or [SKPaymentTransactionStateWrapper.restored].
+  final double transactionTimeStamp;
+
+  /// The unique string identifer of the transaction.
+  ///
+  /// It is only defined when the [transactionState] is [SKPaymentTransactionStateWrapper.purchased] or [SKPaymentTransactionStateWrapper.restored].
+  /// You may wish to record this string as part of an audit trail for App Store purchases.
+  /// The value of this string corresponds to the same property in the receipt.
+  final String transactionIdentifier;
+
+  /// An array of the [SKDownloadWrapper] object of this transaction.
+  ///
+  /// Only available if the transaction contains downloadable contents.
+  ///
+  /// It is only defined when the [transactionState] is [SKPaymentTransactionStateWrapper.purchased].
+  /// Must be used to download the transaction's content before the transaction is finished.
+  final List<SKDownloadWrapper> downloads;
+
+  /// The error object, only available if the [transactionState] is [SKPaymentTransactionStateWrapper.failed].
+  final SKError error;
+}
+
+/// Dart wrapper around StoreKit's [SKDownloadState](https://developer.apple.com/documentation/storekit/skdownloadstate?language=objc).
+enum SKDownloadState {
+  /// Indicates that downloadable content is waiting to start.
+  @JsonValue(0)
+  waiting,
+
+  /// The downloadable content is currently being downloaded
+  @JsonValue(1)
+  active,
+
+  /// The app paused the download.
+  @JsonValue(2)
+  pause,
+
+  /// The content is successfully downloaded.
+  @JsonValue(3)
+  finished,
+
+  /// Indicates that some error occurred while the content was being downloaded.
+  @JsonValue(4)
+  failed,
+
+  /// The app canceled the download.
+  @JsonValue(5)
+  cancelled,
+}
+
+/// Dart wrapper around StoreKit's [SKDownload](https://developer.apple.com/documentation/storekit/skdownload?language=objc).
+///
+/// When a product is created in the App Store Connect, one or more download contents can be associated with it.
+/// When the product is purchased, a List of download object will present in an [SKPaymentTransactionWrapper] object.
+/// Then download objects can be added to the payment queue.
+/// Read the [contentURL] to get the URL of the downloaded content after download complete.A
```

Comment:
nit: "after the download completes."

"download" needs a definite article, and "to complete" needs to be in the singular form of "to complete" [so that it matches the number of downloads](https://en.wikipedia.org/wiki/Agreement_(linguistics)#Number).